theano/tensor/nnet/nnet.py

```diff
@@ -2014,7 +2014,8 @@ def binary_crossentropy(output, target):
     TODO : Rewrite as a scalar, and then broadcast to tensor.
 
     """
-    return -(target * tensor.log(output) + (1.0 - target) * tensor.log(1.0 - output))
+    one = np.asarray(1.0).astype(theano.config.floatX)
```

Comment:
Sorry about the delayed reply. I later recognised there was a problem with my code, that it passed scalars to this method instead of vectors. The problem is when I pass a float32 instead of an arrays/vectors of float32 to this method, it gives me a float64 return. This isn't a trivial case and hence I don't think this difference would matter. I think this is more of a numpy issue, am I correct?